src/vcpkg/binarycaching.cpp

```diff
@@ -211,18 +213,444 @@ namespace
+        };
+        DeletePolicy delete_policy = DeletePolicy::None;
+        int64_t filetime_now(const Filesystem& fs)
+        {
+            switch (delete_policy)
+            {
+                case DeletePolicy::OldestAccessDate: return fs.last_access_time_now();
+                case DeletePolicy::OldestModificationDate: return fs.last_write_time_now();
+                default: Checks::unreachable(VCPKG_LINE_INFO);
+            }
+        }
+        int64_t last_time(const Filesystem& fs, const Path& path, std::error_code& ec)
+        {
+            switch (delete_policy)
+            {
+                case DeletePolicy::OldestAccessDate: return fs.last_access_time(path, ec);
+
+                case DeletePolicy::OldestModificationDate: return fs.last_write_time(path, ec);
+                default: Checks::unreachable(VCPKG_LINE_INFO);
+            }
+        }
+    };
+
+    static StringView to_string(FolderSettings::DeletePolicy policy)
+    {
+        switch (policy)
+        {
+            case FolderSettings::DeletePolicy::None: return "None";
+            case FolderSettings::DeletePolicy::OldestAccessDate: return "OldestAccessDate";
+            case FolderSettings::DeletePolicy::OldestModificationDate: return "OldestModificationDate";
+        }
+        Checks::unreachable(VCPKG_LINE_INFO);
+    }
+
+    struct DeletePolicyDeserializer final : Json::IDeserializer<FolderSettings::DeletePolicy>
+    {
+        LocalizedString type_name() const override { return LocalizedString::from_raw("er"); }
+        Optional<FolderSettings::DeletePolicy> visit_string(Json::Reader& r, StringView value) const override
+        {
+            if (value == "None")
+            {
+                return FolderSettings::DeletePolicy::None;
+            }
+            else if (value == "OldestAccessDate")
+            {
+                return FolderSettings::DeletePolicy::OldestAccessDate;
+            }
+            else if (value == "OldestModificationDate")
+            {
+                return FolderSettings::DeletePolicy::OldestModificationDate;
+            }
+            r.add_generic_error(type_name(), LocalizedString::from_raw("Unexped DeletePolicy"));
+            return nullopt;
+        }
+    };
+
+    struct FolderSettingsDeserializer : Json::IDeserializer<FolderSettings>
+    {
+        LocalizedString type_name() const override { return LocalizedString::from_raw("FolderSettings"); }
+
+        constexpr static StringLiteral MAX_SIZE_GB = "max-size-in-gb";
+        constexpr static StringLiteral MAX_AGE_DAYS = "max-age-in-days";
+        constexpr static StringLiteral KEEP_AVAILABLE_PERCENTAGE = "keep-available-in-percentage";
+        constexpr static StringLiteral DELETE_POLICY = "delete-policy";
+        constexpr static StringLiteral MODIFICATION_DATE_UPDATE_INTERVAL = "modification-date-update-interval";
+
+        Optional<FolderSettings> visit_object(Json::Reader& r, const Json::Object& obj) const override
```

Comment:
```suggestion
        Optional<FolderSettings> visit_object(Json::Reader& reader, const Json::Object& obj) const override
```

I'd prefer a more meaningful name.